[Event Grid] Build test package failing in nightly runs

Event Grid nightly test runs are failing with:

> Error message:
>Executing: "tsc -p ."
eventGridClient.spec.ts(59,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
eventGridClient.spec.ts(59,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
  The types of 'request.body' are incompatible between these types.
    Type 'import("/mnt/vss/_work/1/s/sdk/core/core-rest-pipeline/types/latest/core-rest-pipeline").RequestBodyType | undefined' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-rest-pipeline@1.4.0/node_modules/@azure/core-rest-pipeline/types/latest/core-rest-pipeline").RequestBodyType | undefined'.
      Type 'ReadableStream<Uint8Array>' is not assignable to type 'RequestBodyType | undefined'.
        Type 'ReadableStream<Uint8Array>' is missing the following properties from type 'ReadableStream': readable, read, setEncoding, pause, and 22 more.
eventGridClient.spec.ts(89,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
eventGridClient.spec.ts(167,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
eventGridClient.spec.ts(197,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
eventGridClient.spec.ts(223,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
eventGridClient.spec.ts(315,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
eventGridClient.spec.ts(341,38): error TS2322: Type 'import("/mnt/vss/_work/1/s/sdk/core/core-client/types/latest/core-client").FullOperationResponse' is not assignable to type 'import("/mnt/vss/_work/1/s/common/temp/node_modules/.pnpm/@azure+core-client@1.4.0/node_modules/@azure/core-client/types/latest/core-client").FullOperationResponse'.
The script failed with exit code 2


For more details check here:

- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1315727&view=results
- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1318421&view=results

--- a/sdk/eventgrid/eventgrid/test/public/eventGridClient.spec.ts
+++ b/sdk/eventgrid/eventgrid/test/public/eventGridClient.spec.ts
@@ -12,19 +12,13 @@ import { resetTracer, setTracer } from "@azure/test-utils";
 
 import { AzureKeyCredential, EventGridPublisherClient } from "../../src";
 
-import { FullOperationResponse } from "@azure/core-client";
 import { RestError } from "@azure/core-rest-pipeline";
 import { setSpan, context } from "@azure/core-tracing";
 
 describe("EventGridPublisherClient", function (this: Suite) {
   let recorder: Recorder;
-  let res: FullOperationResponse | undefined;
 
   this.timeout(10000);
-
-  beforeEach(function () {
-    res = undefined;
-  });
 
   describe("#send (EventGrid schema)", function () {
     let client: EventGridPublisherClient<"EventGrid">;
@@ -43,6 +37,8 @@ describe("EventGridPublisherClient", function (this: Suite) {
     });
 
     it("sends a single event", async () => {
+      let status: number | undefined;
+
       await client.send(
         [
           {
@@ -56,13 +52,15 @@ describe("EventGridPublisherClient", function (this: Suite) {
             },
           },
         ],
-        { onResponse: (response) => (res = response) }
+        { onResponse: (response) => (status = response.status) }
       );
 
-      assert.equal(res?.status, 200);
+      assert.strictEqual(status, 200);
     });
 
     it("sends multiple events", async () => {
+      let status: number | undefined;
+
       await client.send(
         [
           {
@@ -86,10 +84,10 @@ describe("EventGridPublisherClient", function (this: Suite) {
             },
           },
         ],
-        { onResponse: (response) => (res = response) }
+        { onResponse: (response) => (status = response.status) }
       );
 
-      assert.equal(res?.status, 200);
+      assert.strictEqual(status, 200);
     });
   });
 
@@ -152,6 +150,8 @@ describe("EventGridPublisherClient", function (this: Suite) {
     });
 
     it("sends a single event", async () => {
+      let status: number | undefined;
+
       await client.send(
         [
           {
@@ -164,13 +164,15 @@ describe("EventGridPublisherClient", function (this: Suite) {
             },
           },
         ],
-        { onResponse: (response) => (res = response) }
+        { onResponse: (response) => (status = response.status) }
       );
 
-      assert.equal(res?.status, 200);
+      assert.strictEqual(status, 200);
     });
 
     it("sends multiple events", async () => {
+      let status: number | undefined;
+
       await client.send(
         [
           {
@@ -194,13 +196,15 @@ describe("EventGridPublisherClient", function (this: Suite) {
             },
           },
         ],
-        { onResponse: (response) => (res = response) }
+        { onResponse: (response) => (status = response.status) }
       );
 
-      assert.equal(res?.status, 200);
+      assert.strictEqual(status, 200);
     });
 
     it("enriches events with distributed tracing information", async () => {
+      let requestBody: string | undefined;
+
       const tracer = setTracer();
       const rootSpan = tracer.startSpan("root");
       await client.send(
@@ -220,13 +224,13 @@ describe("EventGridPublisherClient", function (this: Suite) {
           tracingOptions: {
             tracingContext: setSpan(context.active(), rootSpan),
           },
-          onResponse: (response) => (res = response),
+          onResponse: (response) => (requestBody = response.request.body as string),
         }
       );
 
       rootSpan.end();
 
-      const parsedBody = JSON.parse(res?.request.body as string);
+      const parsedBody = JSON.parse(requestBody || "");
 
       assert.isArray(parsedBody);
       assert.equal(
@@ -301,6 +305,8 @@ describe("EventGridPublisherClient", function (this: Suite) {
     });
 
     it("sends a single event", async () => {
+      let status: number | undefined;
+
       await client.send(
         [
           {
@@ -312,13 +318,15 @@ describe("EventGridPublisherClient", function (this: Suite) {
             },
           },
         ],
-        { onResponse: (response) => (res = response) }
+        { onResponse: (response) => (status = response.status) }
       );
 
-      assert.equal(res?.status, 200);
+      assert.strictEqual(status, 200);
     });
 
     it("sends multiple events", async () => {
+      let status: number | undefined;
+
       await client.send(
         [
           {
@@ -338,10 +346,10 @@ describe("EventGridPublisherClient", function (this: Suite) {
             },
           },
         ],
-        { onResponse: (response) => (res = response) }
+        { onResponse: (response) => (status = response.status) }
       );
 
-      assert.equal(res?.status, 200);
+      assert.strictEqual(status, 200);
     });
   });
 

